game/scripts/vscripts/util.ts

```diff
@@ -455,3 +457,25 @@ export function clearAttachedHighlightParticlesFromUnits(units: CDOTA_BaseNPC[])
         }
     }
 }
+
+/**
+ * Sets the player's respawn settings.
+ * @param respawnLocation The location the player should respawn at.
+ * @param respawnTime How long it should take the player to respawn.
+ */
+export function setRespawnSettings(respawnLocation: Vector, respawnTime: number) {
+    if (respawnListener) {
+        StopListeningToGameEvent(respawnListener)
+        respawnListener = undefined;
+    }
+
+    respawnListener = ListenToGameEvent("entity_killed", event => {
+        const hero = getOrError(getPlayerHero())
```

Comment:
Can remove getOrError here, if it's undefined then they won't be equal (doesn't matter right now, maybe later we'll want to not have a player hero?)